core/src/main/java/org/elasticsearch/common/settings/Settings.java

```diff
@@ -963,6 +964,21 @@ public Builder loadFromSource(String source) {
         }
 
         /**
+         * Loads settings from the actual string content that represents them using the
+         * {@link SettingsLoaderFactory#loaderFromXContentType(XContentType)} method to obtain a loader
+         */
+        public Builder loadFromSource(String source, XContentType xContentType) {
+            SettingsLoader settingsLoader = SettingsLoaderFactory.loaderFromXContentType(xContentType);
+            try {
+                Map<String, String> loadedSettings = settingsLoader.load(source);
```

Comment:
I added a method to XContent to represent whether it is representable as a string. I think a subset might complicate things